Sources/PeripheryKit/Results/OutputDeclarationFilter.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import SystemPackage
+import Shared
+import FilenameMatcher
+
+public final class OutputDeclarationFilter {
+    private let configuration: Configuration
+    private let logger: Logger
+    private let contextualLogger: ContextualLogger
+
+    public required init(configuration: Configuration = .shared, logger: Logger = .init()) {
+        self.configuration = configuration
+        self.logger = logger
+        self.contextualLogger = logger.contextualized(with: "report:filter")
+    }
+
+    public func filter(_ declarations: [ScanResult]) throws -> [ScanResult] {
+        var declarations = declarations
+
+        if let baselinePath = configuration.baseline {
+            let data = try Data(contentsOf: baselinePath.url)
+            let baseline = try JSONDecoder().decode(Baseline.self, from: data)
+            var didFilterDeclaration = false
+            declarations = declarations.filter {
+                let isDisjoint = $0.usrs.isDisjoint(with: baseline.usrs)
```

Comment:
So I think there is a problem with comparing `usrs` directly - in at least a couple of places the `usr` encodes the location, including line and character. For example:

https://github.com/peripheryapp/periphery/blob/7a8b3604a37417b69fa9d694f0bc972bfabe8f71/Sources/PeripheryKit/Syntax/UnusedParameterParser.swift#L68

and

https://github.com/peripheryapp/periphery/blob/7a8b3604a37417b69fa9d694f0bc972bfabe8f71/Sources/PeripheryKit/SourceGraph/SourceGraph.swift#L263

This means that unrelated changes to a file will cause these issues to not match against the baseline because the line numbers will have changed - this is specifically the problem that the original implementation was intended to work around.

Additionally, unless the absolute paths in the baseline match, the violations will not match.

With regard to the latter, I built everything to use relative paths initially, but always intended to do something like respect the `--relative-results` option eventually.

With regard to the former (line and character) you could either remove those from the usrs if that's possible, or if not, then ignore line and character when comparing. That was the intent behind the somewhat hacky conversion in https://github.com/mildm8nnered/periphery/blob/5a40b639019eb920d622faeac134b51da2400072/Sources/PeripheryKit/Baseline.swift#L203

Reply:
Yep, you're spot on. I'm actually working on resolving these two issues as we speak 😄

Reply:
The unused param case is solved by removing the location from the generated USR, and using the USR from the function decl instead.

The unused imports now use relative paths, but I think we still need to include the line & column, as Swift allows duplicate imports. In this small case, changes in the line ordering would bypass the baseline, but I think that's an OK trade-off given the much simpler implementation. Removing unused imports is also trival, so I don't see that being a huge burden.

Reply:
I think that's fair enough - I missed the unused import case in my version because I didn't have any unused imports in my codebase anymore :-)